src/hooks/useMergedRefs.ts

```diff
@@ -1,57 +1,33 @@
-import { useCallback } from 'react';
+import type { RefCallback, Ref } from 'react';
 
 /**
- * NOTE: This was "borrowed" from reach-ui!
+ * Assign a value to a ref function or object.
  */
-
-/**
- * React.Ref uses the readonly type `React.RefObject` instead of
- * `React.MutableRefObject`, We pretty much always assume ref objects are
- * mutable (at least when we create them), so this type is a workaround so some
- * of the weird mechanics of using refs with TS.
- */
-export type AssignableRef<ValueType> =
-  | {
-      bivarianceHack(instance: ValueType | null): void;
-    }['bivarianceHack']
-  | React.MutableRefObject<ValueType | null>;
-
-/**
- * Passes or assigns an arbitrary value to a ref function or object.
- *
- * @param ref
- * @param value
- */
-export function assignRef<RefValueType = any>(
-  ref: AssignableRef<RefValueType> | null | undefined,
-  value: any,
-) {
+function assignRef<Val>(ref: Ref<Val>, value: Val) {
   if (ref == null) return;
   if (typeof ref === 'function') {
     ref(value);
   } else {
-    try {
-      ref.current = value;
-    } catch (error) {
-      throw new Error(`Cannot assign value "${value}" to ref "${ref}"`);
```

Comment:
Don't need the try-catch, since we're handling every possible cause of `Ref<T>`. That's the beauty of static typing.